### ISSUE TITLE
An Analytic Layer-wise Deep Learning Framework with Applications to Robotics. (arXiv:2102.03705v2 [cs.RO] UPDATED)

### ISSUE THREAD

**Issue by CoffeeKumazaki** [CoffeeKumazaki/arXiv#27406]
An Analytic Layer-wise Deep Learning Framework with Applications to Robotics. (arXiv:2102.03705v2 [cs.RO] UPDATED)<br>
https://ift.tt/FaJdYHD<br>
<p>Deep learning (DL) has achieved great success in many applications, but it
has been less well analyzed from the theoretical perspective. The unexplainable
success of black-box DL models has raised questions among scientists and
promoted the emergence of the field of explainable artificial intelligence
(XAI). In robotics, it is particularly important to deploy DL algorithms in a
predictable and stable manner as robots are active agents that need to interact
safely with the physical world. This paper presents an analytic deep learning
framework for fully connected neural networks, which can be applied for both
regression problems and classification problems. Examples for regression and
classification problems include online robot control and robot vision. We
present two layer-wise learning algorithms such that the convergence of the
learning systems can be analyzed. Firstly, an inverse layer-wise learning
algorithm for multilayer networks with convergence analysis for each layer is
presented to understand the problems of layer-wise deep learning. Secondly, a
forward progressive learning algorithm where the deep networks are built
progressively by using single hidden layer networks is developed to achieve
better accuracy. It is shown that the progressive learning method can be used
for fine-tuning of weights from convergence point of view. The effectiveness of
the proposed framework is illustrated based on classical benchmark recognition
tasks using the MNIST and CIFAR-10 datasets and the results show a good balance
between performance and explainability. The proposed method is subsequently
applied for online learning of robot kinematics and experimental results on
kinematic control of UR5e robot with unknown model are presented.
</p>
<br>
<br>
via cs.RO updates on arXiv.org http://arxiv.org/<br>
